sycl/source/detail/force_device.cpp

```diff
@@ -21,18 +22,22 @@ bool match_types(const info::device_type &l, const info::device_type &r) {
 
 info::device_type get_forced_type() {
   if (const char *val = std::getenv("SYCL_DEVICE_TYPE")) {
-    if (string_class(val) == "CPU") {
+    std::string type(val);
+    std::transform(type.begin(), type.end(), type.begin(), ::tolower);
+
+    if (type == "cpu") {
       return info::device_type::cpu;
     }
-    if (string_class(val) == "GPU") {
+    if (type == "gpu") {
       return info::device_type::gpu;
     }
-    if (string_class(val) == "ACC") {
+    if (type == "acc") {
       return info::device_type::accelerator;
     }
-    if (string_class(val) == "HOST") {
+    if (type == "host") {
       return info::device_type::host;
     }
+    throw cl::sycl::runtime_error("SYCL_DEVICE_TYPE is not available.  Must be GPU, CPU, ACC or HOST");
```

Comment:
"SYCL_DEVICE_TYPE is not available." sounds confusing, what about:
"SYCL_DEVICE_TYPE value is not recognized." ?

Reply:
Done.